lambdas/tests/unit/services/jwt-verifier.test.ts

```diff
@@ -0,0 +1,211 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { JwtVerificationConfig } from "../../../src/common/config/jwt-verification-config";
+import { JwtVerifier } from "../../../src/common/security/jwt-verifier";
+import * as jose from 'jose'
+import { importJWK, JWTHeaderParameters, jwtVerify } from "jose";
+import { JwkKeyExportOptions } from "crypto";
+jest.mock("jose", () => ({
+    importJWK: jest.fn(),
+    jwtVerify: jest.fn(),
+}));
+
+describe("JwtVerifier", () => {
+    const logger = {
```

Comment:
You don't want to assign variables outside of a `before` / `beforeEach` block as you will end up with weird errors due to how the tests are instantiated

Define the variables using a `let` in the block so they can be reused, and then assign them in a `beforeEach` - that way the value is guaranteed to remain consistent between tests.

Basically, the testing framework runs all the code as it's assembling the tests. If it's normal code its run instantly, but if its Jest functions (e.g `describe` / ` beforeEach`) it'll still run instantly, but the Jest function will create a hook using the internal function to be run later